File: 프로그래머스/2/389480. 완전범죄/완전범죄.java
import java.util.Arrays;

class Solution {
    public int solution(int[][] info, int n, int m) {
        Arrays.sort(info, (x, y) -> {
            int diff = (x[1] - x[0]) - (y[1] - y[0]);
            return diff != 0 ? diff : Integer.compare(y[0], x[0]);
        });
        
        int aSum = 0, bSum = 0;
        
        for (int[] pair : info) {
            int a = pair[0], b = pair[1];
            
            if (bSum + b < m) {
                bSum += b;
            } else if (aSum + a < n) {
                aSum += a;
            } else {
                return -1;
            }
        }
        
        return aSum;
    }
}

File: 프로그래머스/2/389480. 완전범죄/완전범죄.py
'''
1. info 정렬 (첫 번째 인덱스 높은순, 두 번째 인덱스 낮은순) -> nlogn
2. 앞에서부터 도둑이 가능한만큼 훔치기 -> n
'''
def solution(info: list, n: int, m: int) -> int:
    answer = [0, 0]
    
    info.sort(key=lambda x: (x[1] - x[0], -x[0]))
    
    for a, b in info:
        if answer[1] + b < m:
            answer[1] += b
        elif answer[0] + a < n:
            answer[0] += a
        else:
            return -1
            
    return answer[0]